network/network.go

```diff
@@ -303,6 +303,21 @@ func (i *InterfaceInfo) CIDRAddress() string {
 	return ipNet.String()
 }
 
+// MAC returns the MacAddress member value.
```

Comment:
Hmm, it's weird that you had to come up with slightly different names so that the accessors needed for the interface didn't collide with the public attributes. That suggests that this isn't the right thing.

Did you really need an interface for `InterfaceInfo` for testing? The structs in network are passive data structures and can be directly constructed, so I think it would make more sense to use them for testing. Is there some reason you can't?